crates/bevy_ui/src/widget/text.rs

```diff
@@ -123,6 +131,12 @@ pub fn text_system(
                     };
                 }
             }
+
+            if let Some(mut bidi_corrected) = bidi_corrected {
+                *bidi_corrected = bidi_corrected_internal;
+            } else {
+                commands.entity(entity).insert(bidi_corrected_internal);
+            }
```

Comment:
Why do we need this logic in two places? Would it be possible to safely separate this in a new system and to then assume that bidi corrected text is available so that we can only query for that?